iquip/apps/monitor.py

```diff
@@ -1011,6 +1114,31 @@ def _setDDSSwitch(self, device: str, channel: int, on: bool):
         self.ddsSwitchThread.finished.connect(self.ddsSwitchThread.deleteLater)
         self.ddsSwitchThread.start()
 
+    @pyqtSlot(dict)
+    def _updateTTLStatus(self, modifications: Dict[str, Dict[str, bool]]):
+        """Updates the TTL status.
+        
+        Args:
+            See _TTLStatusThread signals section.
+        """
+        for device, output in modifications["probe"].items():
+            name = self.ttlToName[device]
+            self.ttlControllerFrame.ttlWidgets[name].outputChanged.emit(output)
+        for device, level in modifications["level"].items():
+            name = self.ttlToName[device]
+            self.ttlControllerFrame.ttlWidgets[name].levelChanged.emit(level)
+        for device, override in modifications["override"].items():
+            name = self.ttlToName[device]
+            self.ttlControllerFrame.ttlWidgets[name].overrideChanged.emit(override)
+
+    def _startTTLStatusThread(self):
+        """Creates and starts a new _TTLStatusThread instance."""
+        devices = list(self.ttlToName)
+        self.ttlStatusThread = _TTLStatusThread(self.proxy_ip, self.proxy_port, devices)
+        self.ttlStatusThread.fetched.connect(self._updateTTLStatus, type=Qt.QueuedConnection)
+        self.ttlStatusThread.finished.connect(self.ttlStatusThread.deleteLater)
```

Comment:
Let's support retrying connection in case the connection is lost.
You can separate it as another issue if you want.

Reply:
The size of this PR already became too large, so I will separate it as another issue.